cmd/all-in-one/main.go

```diff
@@ -272,3 +296,31 @@ func initTracer(metricsFactory metrics.Factory, logger *zap.Logger) io.Closer {
 	opentracing.SetGlobalTracer(tracer)
 	return closer
 }
+
+// metricsQueryEnabled returns whether if metrics querying capabilities are enabled.
+// To avoid introducing a breaking change, the Metrics Querying feature must be
+// explicitly opted-in by setting the METRICS_STORAGE_TYPE env var.
+// An unset METRICS_STORAGE_TYPE will disable this feature.
+func metricsQueryEnabled(fc metricsPlugin.FactoryConfig) bool {
+	return fc.MetricsStorageType != ""
+}
+
+func createMetricsReaderFactory(fc metricsPlugin.FactoryConfig) (*metricsPlugin.Factory, error) {
+	if !metricsQueryEnabled(fc) {
```

Comment:
are these checks still necessary? wouldn't you always get _something_ back (perhaps disabled service)?

Reply:
Yeah, it's still necessary otherwise: `Cannot initialize metrics store factory: unknown metrics type "". Valid types are [prometheus]`

Do you think it's worth doing something similar with the factories? i.e. have `disabledMetricsReaderFactory` that simply returns a `nil` Reader? That way, these checks won't be necessary.

Reply:
I think you can extend the same approach - consider default metrics store type to be "none" and return a "disabled reader" in that case. Basically push this all the way down (you won't need "disabled service" in this case).